python/lsst/pipe/tasks/mocks/mockCoadd.py

```diff
@@ -277,7 +277,8 @@ def buildMockCoadd(self, butler, truthCatalog=None, skyMap=None, tract=0):
         tractInfo = skyMap[tract]
         tractWcs = tractInfo.getWcs()
         for patchRef in self.iterPatchRefs(butler, tractInfo):
-            exposure = patchRef.get(self.config.coaddName + "Coadd")
+            # TODO: pybind11 remove `immediate=True` once DM-9112 is resolved
```

Comment:
Leaving aside the question of original intent, I think it's good to have tests that use `immediate=False` in order to cover the case where the object is a proxy. Hence why I feel this commit should be treated as a workaround rather than a permanent change.